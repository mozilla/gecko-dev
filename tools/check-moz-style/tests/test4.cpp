class ShouldUseExplicit {
  // runtime/explicit
  ShouldUseExplicit(int i);
};

// readability/function
int foo(int) {
}

int main() {
  int i = 0;

  // readability/control_flow
  // XXX This doesn't trigger it. It needs to be fixed.
  if (i) {
    return;
  } else {
    i++;
  }

  // whitespace/parens
  if(i){}

  // readability/casting
  void* bad = (void*)i;

  // readability/comparison_to_zero
  if (i == true) {}
  if (i == false) {}
  if (i != true) {}
  if (i != false) {}
  if (i == NULL) {}
  if (i != NULL) {}
  if (i == nullptr) {}
  if (i != nullptr) {}
  if (i) {}
  if (!i) {}

  return 0;
}
