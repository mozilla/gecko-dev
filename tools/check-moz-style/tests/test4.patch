# Test
Bug 12 - patch summary with no bug number

Some bogus patch description
diff --git a/tests/test4.cpp b/tests/test4.cpp
new file mode 100644
--- /dev/null
+++ b/tests/test4.cpp
@@ -0,0 +1,49 @@
+class ShouldUseExplicit {
+  // runtime/explicit
+  ShouldUseExplicit(int i);
+};
+
+// readability/function
+int foo(int) {
+}
+
+int main() {
+  int i = 0;
+
+  // readability/control_flow
+  // XXX This doesn't trigger it. It needs to be fixed.
+  if (i) {
+    return;
+  } else {
+    i++;
+  }
+
+  // whitespace/parens
+  if(i){}
+
+  // readability/casting
+  void* bad = (void*)i;
+
+  // readability/comparison_to_zero
+  if (i == true) {}
+  if (i == false) {}
+  if (i != true) {}
+  if (i != false) {}
+  if (i == NULL) {}
+  if (i != NULL) {}
+  if (i == nullptr) {}
+  if (i != nullptr) {}
+  if (i) {}
+  if (!i) {}
+
+  return 0;
+}
