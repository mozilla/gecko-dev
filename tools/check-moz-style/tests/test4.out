tests/test4.cpp:1:  No copyright message found.  [legal/copyright] [3]
tests/test4.cpp:3:  Single-argument constructors should be marked explicit.  [runtime/explicit] [5]
tests/test4.cpp:7:  All parameters should be named in a function  [readability/function] [3]
tests/test4.cpp:22:  Missing space before ( in if(  [whitespace/parens] [5]
tests/test4.cpp:22:  Missing space before {  [whitespace/braces] [5]
tests/test4.cpp:25:  Using C-style cast.  Use reinterpret_cast<void*>(...) instead  [readability/casting] [4]
tests/test4.cpp:28:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
tests/test4.cpp:29:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
tests/test4.cpp:30:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
tests/test4.cpp:31:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
tests/test4.cpp:32:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
tests/test4.cpp:33:  Tests for true/false, null/non-null, and zero/non-zero should all be done without equality comparisons.  [readability/comparison_to_zero] [5]
Done processing tests/test4.cpp
